### PR TITLE
[html-replace-webpack-plugin] Adding initial types for `html-replace-webpack-plugin`

### PR DESCRIPTION
**Pull request by bricka**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/html-replace-webpack-plugin/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-unnecessary-class": false
+    }
+}
```
Comment (peterblazejewicz) [inline types/html-replace-webpack-plugin/tslint.json:6]:
```suggestion
{
    "extends": "dtslint/dt.json"
}
```
can this be replaced with inline exlclusion? (there is only one place), thx!
https://github.com/DefinitelyTyped/DefinitelyTyped#linter-tslintjson

**Review by peterblazejewicz — COMMENTED** [review]
Are you sure it would work without using `apply`? Can you verify this within setup with `@types/webpack` or native types?
```text
Property 'apply' is missing in type 'HtmlReplaceWebpackPlugin' but required in type 'Plugin'.ts(2741)
````

**Comment by bricka** [comment]
@peterblazejewicz Good catch on the `apply`: I don't know why I didn't see that when I was testing. I can't run `npm test` locally right now, unfortunately: I get a bunch of errors about some types not being available locally, and I don't know how to fix them :(.

```
Error: Errors in typescript@4.3 for external dependencies:
../uglify-js/index.d.ts(9,30): error TS7016: Could not find a declaration file for module 'source-map'. '/home/abrick/src/DefinitelyTyped/node_modules/source-map/source-map.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/source-map` if it exists or add a new declaration (.d.ts) file containing `declare module 'source-map';`
```

Let me know if you have any ideas.

**Comment by bricka** [comment]
Although, it looks like the tests passed in the CI. :tada:

**Comment by peterblazejewicz** [comment]
> Although, it looks like the tests passed in the CI. 🎉

The CI diffs the packages and install dependencies for all referenced types. Locally, this is not happening, so one needs to individually install `npm i --no-package-lock` for a `@types/....`, which uses `package.json` defined dependencies. Yes, easier on CI

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@bricka thx!

**Pull request merged**